lib/_http_outgoing.js

```diff
@@ -337,6 +337,14 @@ OutgoingMessage.prototype.getHeader = function(name) {
 };
 
 
+OutgoingMessage.prototype.getAllHeaders = function() {
+  if (!this._headers)
+    return;
+  else
+    return util._extend({}, this._headers);
```

Comment:
Do you think it might be worth using `Object.create(null)` instead of `{}` to avoid extraneous properties on the object?